DSharpPlus.Commands/Invocation/CommandEmitUtil.cs

```diff
@@ -19,106 +19,122 @@ internal static class CommandEmitUtil
     /// <exception cref="InvalidOperationException">Thrown if the command returns anything but ValueTask and Task.</exception>
     public static Func<object?, object?[], ValueTask> GetCommandInvocationFunc(MethodInfo method, object? target)
     {
-        // we assume this is an anonymous method of any sort
+        // This method is very likely to be an anonymous delegate, which happens to be very slow to invoke.
+        // We're going to take the slow path here and simply do `MethodInfo.Invoke`, for lack of a better way.
         if (method.Name.Contains('<') && method.Name.Contains('>') && method.GetCustomAttribute<CompilerGeneratedAttribute>() is not null)
         {
             return AnonymousDelegateUtil.GetAnonymousInvocationFunc(method, target);
         }
-
-        if (method.ReturnType == typeof(ValueTask))
+        else if (method.ReturnType == typeof(ValueTask))
         {
             return GetValueTaskFunc(method);
         }
-
         else if (method.ReturnType == typeof(Task))
         {
             return GetTaskFunc(method);
         }
 
+        // This could happen for `void` methods when the user explicitly builds a command tree with them.
+        // We probably don't want to support `void` since the user will never be able to respond to the command...
+        // Unless if it was done through context checks... Hm.
```

Comment:
we shouldn't encourage misuse of context checks in the name of an already immensely questionable code pattern, ideally